db/builder.cc

```diff
@@ -175,12 +175,17 @@ Status BuildTable(
                                   blob_file_additions)
             : nullptr);
 
+    std::shared_ptr<Logger> info_log;
```

Comment:
We did log using `db_options_.info_log`. Maybe should pass that to `BuildTable`?

Reply:
What I mean is that weirdly enough, it seems we haven't passed the logger to `CompactionIterator` historically (i.e. your code, which passes an empty `shared_ptr`, is functionally equivalent to the checked-in code). But yeah, it might be a good idea to improve this and start passing `db_options_.info_log` like you suggested.

Reply:
I find it weird too. Let me fix it.